ocean.js

```diff
@@ -0,0 +1,251 @@
+
+let fetchArray = [oneOfFive, twoOfFive, threeOfFive, fourOfFive, fiveOfFive];
+
+const token = "WVEwvCMCqAKfcjKwwjLJonCdawPQuVeE";
+// MASTERFETCH FUNCTION CALLS ARRAYFETCH FUNCTION FOR (FETCHARRAY ARRAY)
+
+// ARRAY FETCH FUNCTION CALLED IN MASTERFETCH FUNCTION
+
+function arrayFetch(element) {
+  return fetch(element, {
+    method: "GET",
+    headers: { token: token },
+  })
+    .then((response) => {
+      if (response.ok) {
+        console.log("Successful");
+
+        return response;
+      } else {
+        console.log(response.status);
+      }
+    })
+    .then((response) => response.json())
+    .then((data) => {
+      let WeatherArray;
+      WeatherArray = JSON.parse(localStorage.getItem("WeatherData"));
+      if (WeatherArray === null) {
+        WeatherArray = [];
+        WeatherArray.push(data);
+      } else {
+        WeatherArray.push(data);
+      }
+      localStorage.setItem("WeatherData", JSON.stringify(WeatherArray));
+    });
+}
+const fetchingURL = fetchArray.map((url) => fetch(url).then());
+setTimeout(function () {
+  Promise.all(fetchingURL).then(dataFormat);
+}, 4000);
+localStorage.removeItem("WeatherData");
+function masterFetch() {
+  fetchArray.forEach((element) => arrayFetch(element));
+}
+masterFetch();
+
+//   DATA CONVERSION FOR CHART.JS
+
+function dataFormat() {
+  let WeatherArray = JSON.parse(localStorage.getItem("WeatherData"));
+
+  let ArrayOne = WeatherArray[0]["results"];
+  let ArrayTwo = WeatherArray[1]["results"];
+  let ArrayThree = WeatherArray[2]["results"];
+  let ArrayFour = WeatherArray[3]["results"];
+  let ArrayFive = WeatherArray[4]["results"];
+
+  let MainArray = ArrayOne.concat(ArrayTwo)
+    .concat(ArrayThree)
+    .concat(ArrayFour)
+    .concat(ArrayFive);
+
+  let NOE00134778 = MainArray.filter(
+    ({ station }) => station === "GHCND:NOE00134778"
+  );
+  let GME00111445 = MainArray.filter(
+    ({ station }) => station === "GHCND:GME00111445"
+  );
+  let SPE00120458 = MainArray.filter(
+    ({ station }) => station === "GHCND:SPE00120458"
+  );
+  let USW00012836 = MainArray.filter(
+    ({ station }) => station === "GHCND:USW00012836"
+  );
+  let USW00022521 = MainArray.filter(
+    ({ station }) => station === "GHCND:USW00022521"
+  );
+  let USW00026533 = MainArray.filter(
+    ({ station }) => station === "GHCND:USW00026533"
+  );
+
+  let stationArray = [
+    NOE00134778,
+    GME00111445,
+    SPE00120458,
+    USW00012836,
+    USW00022521,
+    USW00026533,
+  ];
+
+  for (let i = 0; i < stationArray.length; i++) {
+    for (let j = 0; j < stationArray[i].length; j++) {
+      let value = stationArray[i][j]["date"];
+
+      value = value.slice(0, 4);
+
+      value = Number(value);
+
+      stationArray[i][j]["date"] = value;
+    }
+    stationArray[i].sort(function (a, b) {
+      return a.date - b.date;
+    });
+  }
+
+  let NOE00134778date = [];
+  let NOE00134778value = [];
```

Comment:
Could this be an object? Since you're using `...date` and `...value` as variablename.

Reply:
Not sure what you meen? destructured data for Chart.js